tests/conftest.py

```diff
@@ -936,8 +936,14 @@ def mock_hass_config(
 ) -> Generator[None, None, None]:
     """Fixture to mock the content of main configuration.
 
-    Patches homeassistant.config.load_yaml_config_file with `hass_config` parameterized as content.
+    Patches homeassistant.config.load_yaml_config_file and hass.config_entries
+    with `hass_config` as parameterized.
     """
+    if hass_config:
+        hass.config_entries = ConfigEntries(
+            hass,
+            hass_config,
+        )
```

Comment:
Small style suggestion:
```suggestion
        hass.config_entries = ConfigEntries(hass, hass_config)
```